goon/browserassets/js/browserOutput.js

```diff
@@ -871,32 +871,73 @@ $(function() {
 		setCookie('pingdisabled', (opts.pingDisabled ? 'true' : 'false'), 365);
 	});
 
-	$('#saveLog').click(function(e) {
-		var saved = '';
+	$('#saveLog').click(function(e) {	//note that this will only work in browsers supporting CORS (IE 8+ and literally every alternative)
```

Comment:
If it wouldn't be too difficult - could you keep the changes you made originally, and make a `useHastebin` boolean in `opts`? So, if a server operator doesn't want to be bothered with setting up hastebin, chat log saving isn't completely disabled?